scripts/validateTypescriptDeclarations.mts

```diff
@@ -1,28 +1 @@
-/* eslint-disable no-console */
-import { globbySync } from 'globby';
-import fs from 'fs';
-
-/**
- * Validates if there are no missing exports from TS files that would
- * result in an import from a local file.
- */
-function validateFiles() {
-  const declarationFiles = globbySync(['packages/*/build/**/*.d.ts'], {
-    followSymbolicLinks: false,
-  });
-  const invalidFiles = declarationFiles.filter((file) => {
-    const content = fs.readFileSync(file, 'utf8');
-    const regex = /import\(["']packages\//gm;
```

Comment:
🤔 I feel like this is trying to do the exact same thing as testBuiltTypes. Should probably do a bit of an audit on all these scripts.

Reply:
This script was created to prevent issues such as https://github.com/mui/material-ui/issues/33321 (caused by referencing a type that's not exported).

Reply:
🤔 I'm suspecting the root cause of that issue is the use of [`baseUrl`](https://github.com/mui/base-ui/blob/45c8bd1102c66dff234feebc06bb253790a6db5d/tsconfig.base.json#L11). We should try if we can still reproduce this bug with that option removed.